gui/src/renderer/components/Marquee.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import { Animated, Component, Styles, Types, UserInterface, View } from 'reactxp';
+
+const styles = {
+  text: Styles.createTextStyle({
+    // @ts-ignore
+    width: 'fit-content',
+    whiteSpace: 'nowrap',
+  }),
+};
+
+interface IMarqueeProps {
+  style?: Types.StyleRuleSetRecursive<Types.ButtonStyleRuleSet>;
+}
+
+export default class Marquee extends Component<IMarqueeProps> {
+  private initialLeft = Animated.createValue(0.0);
+  private textAnimation = Styles.createAnimatedTextStyle({ left: this.initialLeft });
+  private textRef = React.createRef<Animated.Text>();
+
+  public componentDidMount() {
+    this.startAnimation();
+  }
+
+  public componentDidUpdate() {
+    this.startAnimation();
+  }
+
+  public render() {
+    return (
+      <View>
+        <Animated.Text
+          ref={this.textRef}
+          style={[styles.text, this.textAnimation, this.props.style]}>
+          {this.props.children}
+        </Animated.Text>
+      </View>
+    );
+  }
+
+  private async startAnimation() {
+    setTimeout(async () => {
```

Comment:
Save the timeout and clear it in `componentWillUnmount()` to make sure we don't run anything after the component is gone.